### ISSUE TITLE
The exported DDL of the table for POSTGRES_15 dialect is missing a default value

### ISSUE THREAD

**Issue by mgramin** [jOOQ/jOOQ#15039]
### Expected behavior

I have a simple table in my database with a generated `id` column by default:

```sql
create table "public"."test_test" (
  "id" bigint generated by default as identity not null,
  constraint "test_test_pkey" primary key ("id")
)
```
I expect that jOOQ exports the table's DDL, including the generated by default value.

### Actual behavior

However, I received the table DDL without default values, which resulted in errors during insertion:

```sql
create table "public"."test_test" (
  "id" bigint not null,
  constraint "test_test_pkey"
    primary key ("id")
)
```

### Steps to reproduce the problem

1. Create the simple table with generated by default `id` column in DB:
```sql
create table "public"."test_test" (
  "id" bigint generated by default as identity not null,
  constraint "test_test_pkey" primary key ("id")
)
```

2. Try to export DDL for this table for `POSTGRES_15` dialect:
```java
        Connection conn = DriverManager.getConnection("jdbc:postgresql://localhost:5432/api_postgres", "postgres", "postgres");
        Configuration configuration = new DefaultConfiguration().set(conn).set(SQLDialect.POSTGRES_15);
        Meta currentMeta = using(configuration).meta();

        List<String> queries = Arrays.stream(currentMeta
                        .filterSchemas(v -> v.getName().equalsIgnoreCase("public"))
                        .ddl()
                        .queries())
                .map(Object::toString).toList();

        System.out.println(queries);
```

3. And make sure that DDL doesn't contain a default value:
```sql
create table "public"."test_test" (
  "id" bigint not null,
  constraint "test_test_pkey"
    primary key ("id")
)
```
 
4. But with `POSTGRES_14` dialect it works well:
```java
Configuration configuration = new DefaultConfiguration().set(conn).set(SQLDialect.POSTGRES_15);
```
```sql
create table "public"."test_test" (
  "id" bigint generated by default as identity not null,
  constraint "test_test_pkey" primary key ("id")
)
```

### jOOQ Version

jOOQ Professional Edition 3.18.3

### Database product and version

PostgreSQL 15.2 (Debian 15.2-1.pgdg110+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 10.2.1-6) 10.2.1 20210110, 64-bit

### Java Version

openjdk 17.0.2 2022-01-18

### OS Version

Microsoft Windows [Version 10.0.19044.2846]

### JDBC driver name and version (include name if unofficial driver)

org.postgresql:postgresql:42.6.0

**Comment by lukaseder** [comment]
Thanks a lot for your report. I can see that the `switch` over the dialect version in `Tools::toSQLDDLTypeDeclarationIdentityBeforeNull` doesn't cover `POSTGRES_15`. Unfortunately, our integration tests don't currently run with that dialect *version*, only the the dialect family (which corresponds to the latest version), which is why this wasn't discovered.

I'll fix it right away. The 3.18.4 patch release should be released this week. As a workaround, you can use `SQLDialect.POSTGRES` instead of `SQLDialect.POSTGRES_15`.

**Comment by lukaseder** [comment]
Fixed for jOOQ 3.19.0 and 3.18.4 (https://github.com/jOOQ/jOOQ/issues/15041).

Thanks again for your report!

**Comment by mgramin** [comment]
I think this is one of the quickest bug fixes in jOOQ that I've ever seen, thank you!

PS
Could you tell me, please, where I can receive notifications about the release of the new 3.18.4 version?
